src/system/SystemLayerImplSelect.cpp

```diff
@@ -685,16 +695,21 @@ void LayerImplSelect::HandleEvents()
 
     for (auto & w : mSocketWatchPool)
     {
-        if (w.mFD != kInvalidFd)
+        if (w.mFD != kInvalidFd && w.mCallback != nullptr)
```

Comment:
Hmm.  Wouldn't we want to drain the events even if there is no callback?

Reply:
`SocketEventsFromFDs` doesn't drain anything, it just sets a bit mask based on which of the socket sets then given FD appears in. So this just avoids a bit of unnecessary work.